tests/test_TFNetworkLayer.py

```diff
@@ -5456,6 +5456,53 @@ def test_GatherLayer_broadcast_dim():
   })
 
 
+def test_GatherLayer_batch_dim():
+  from returnn.tf.util.data import batch_dim
+  with make_scope() as session:
+    import numpy as np
+    net = TFNetwork(extern_data=ExternData())
+    b_dim, time_dim, feature_dim = 3, 4, 2
+    # [B, T, F]
+    random = np.random.RandomState(42)
+    values_seqs = random.rand(b_dim, time_dim, feature_dim).astype('float32')
+    values_size = np.array([4, 2, 3])
+    values_placeholder = tf.constant(values_seqs, dtype=tf.float32)
+    values_size_placeholder = {0: tf.constant(values_size, dtype=tf.int32)}
+    values = InternalLayer(
+      name="values", network=net,
+      output=Data(
+        name="values",
+        batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=2,
+        shape=[None, feature_dim],
+        placeholder=values_placeholder,
+        size_placeholder=values_size_placeholder,
+        dim_tags=(batch_dim, SpatialDim("time"), FeatureDim("feature", feature_dim)),
+      ))
+    position_np = np.array([0, 2])
+    position = InternalLayer(
+      name="position", network=net,
+      output=Data(
+        name="position",
+        placeholder=tf.constant(position_np, dtype=tf.int64),
+        batch_dim_axis=0, shape=[], dtype="int64",
+      ))
```

Comment:
@albertz do I need to change the creation of `position` in order to make it have a different batch axis dim tag here?

Reply:
Yes. It's actually not so simple because of the special treatment of the batch dim tag. I'm not sure it's really possible currently.

In practice, in your real code, how would you end up with `position`?

Reply:
> In practice, in your real code, how would you end up with position?
Do you mean what dim tag I get there?
```
>>> position.output.dim_tags[0].description                                                                                                                                                                                     
'batch:position'
```
So it's not actually the global batch dim. I was just confused because I got
```
>>> position.output.dim_tags[0] == values.output.dim_tags[0]                                                                                                                                                                    
True
```
but this is because the check does not cover this case, see comment here: https://github.com/rwth-i6/returnn/pull/1089#discussion_r955964986